Allow for opening maps from the command line

zurgeg opened the pull request:
This PR adds support for
- Drag-n-dropping worlds onto Amulet
- Opening worlds from the CLI (i.e., allowing apps like [MultiMC](https://github.com/MultiMC/Launcher/issues/3713) to open worlds automatically)

Todo:
- [x] Test latest commit
- [x] Check for accidentally committed files 
- [x] Run black

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Amulet-Team/Amulet-Map-Editor?pullRequest=725) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Amulet-Team/Amulet-Map-Editor?pullRequest=725) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Amulet-Team/Amulet-Map-Editor?pullRequest=725) it.</sub>

gentlegiantJGC commented:
I would like to support opening worlds from the command line.
This would be better implemented in the `__main__.py` module as an `else` block at the end and using argparse.
Perhaps the flag `--open-level path`. I am hesitant to use positional arguments in case our needs change.

https://github.com/Amulet-Team/Amulet-Map-Editor/blob/0.9/amulet_map_editor/__main__.py#L36-L43

```py
        try:
            app = AmuletApp(0)
            app.MainLoop()
        except Exception as e:
            log.critical(
                f"Amulet Crashed. Sorry about that. Please report it to a developer if you think this is an issue. \n{traceback.format_exc()}"
            )
            input("Press ENTER to continue.")
        else:
            # run argparse
``` 

In 0.10 I have added a method to the app instance to open and close levels which is what you would need to call.
<https://github.com/Amulet-Team/Amulet-Map-Editor/blob/0.10/amulet_map_editor/api/framework/app.py#L27-L34>

Is there need for the Pipfile? I have updated the packaging system to use the pyproject.toml and setup.cfg workflow and removed the requirements.txt file as this was quite outdated. Is there anything that would need a Pipfile that couldn't work through the pyproject.toml file?

I am going to close this pull request because the changes I have requested are effectively a full rewrite. If you want to make the changed please open another pull request.

zurgeg commented:
> I would like to support opening worlds from the command line. This would be better implemented in the `__main__.py` module as an `else` block at the end and using argparse. Perhaps the flag `--open-level path`. I am hesitant to use positional arguments in case our needs change.
> 
> https://github.com/Amulet-Team/Amulet-Map-Editor/blob/0.9/amulet_map_editor/__main__.py#L36-L43
> 
> ```python
>         try:
>             app = AmuletApp(0)
>             app.MainLoop()
>         except Exception as e:
>             log.critical(
>                 f"Amulet Crashed. Sorry about that. Please report it to a developer if you think this is an issue. \n{traceback.format_exc()}"
>             )
>             input("Press ENTER to continue.")
>         else:
>             # run argparse
> ```
> 
> In 0.10 I have added a method to the app instance to open and close levels which is what you would need to call. https://github.com/Amulet-Team/Amulet-Map-Editor/blob/0.10/amulet_map_editor/api/framework/app.py#L27-L34
> 
> Is there need for the Pipfile? I have updated the packaging system to use the pyproject.toml and setup.cfg workflow and removed the requirements.txt file as this was quite outdated. Is there anything that would need a Pipfile that couldn't work through the pyproject.toml file?
> 
> I am going to close this pull request because the changes I have requested are effectively a full rewrite. If you want to make the changed please open another pull request.

the pipfile was an accident :P
The reason I chose to use positional arguments was to make MMC support easier without needing much modification. On top of that, unless we use positional arguments, we can't have drag-n-drop support, but feel free to make your own decisions regarding this. Thanks for taking a look at this.
